### PR TITLE
kernel: Use CONFIG_KSU=m to compile lkm

### PR DESCRIPTION
**Pull request by dabao1955**
before:
CONFIG_KSU=y
CONFIG_KSU_MODULE=y

after:
CONFIG_KSU=m

### CODE REVIEW

**Comment by tiann** [comment]
here: https://github.com/tiann/KernelSU/blob/main/.github/workflows/gki-kernel.yml#L169

**Comment by dabao1955** [comment]
fixed

File: kernel/Makefile [hunk 1]
```diff
@@ -17,7 +17,7 @@ ccflags-y += -I$(srctree)/security/selinux -I$(srctree)/security/selinux/include
 ccflags-y += -I$(objtree)/security/selinux -include $(srctree)/include/uapi/asm-generic/errno.h
 
 ifndef KSU_MODULE
-obj-y += kernelsu.o
+obj-$(CONFIG_KSU) += kernelsu.o
```
Comment (tiann) [inline kernel/Makefile:20]: Maybe you can use `obj-$(CONFIG_KSU) += kernelsu.o` and then modify the build-lkm.yml?

Comment (tiann) [reply kernel/Makefile:20]: And then we can delete KSU_MODULE

**Review by tiann — CHANGES_REQUESTED** [review]

**Comment by dabao1955** [comment]
well bazel does not work:
```
[411 / 423] Creating kernel config (lto=thin;trim) @//common:kernel_aarch64_config; 0s local
ERROR: /home/runner/work/KernelSU/KernelSU/android-kernel/common/BUILD.bazel:115:22: Creating kernel config (lto=thin;trim) @//common:kernel_aarch64_config failed: (Exit 1): bash failed: error executing KernelConfig command (from target //common:kernel_aarch64_config) /bin/bash -c ... (remaining 1 argument skipped)
--- common/arch/arm64/configs/gki_defconfig	2024-03-18 06:11:25.782582331 +0000
+++ /home/runner/work/KernelSU/KernelSU/android-kernel/out/cache/8bda6b74/common/defconfig	2024-03-18 06:11:59.610924758 +0000
@@ -592,6 +592,7 @@
 CONFIG_LIBNVDIMM=y
 # CONFIG_ND_BLK is not set
 CONFIG_INTERCONNECT=y
+CONFIG_KSU=m
 CONFIG_EXT4_FS=y
 CONFIG_EXT4_FS_POSIX_ACL=y
 CONFIG_EXT4_FS_SECURITY=y
@@ -738,4 +739,3 @@
 CONFIG_HIST_TRIGGERS=y
 CONFIG_PID_IN_CONTEXTIDR=y
 # CONFIG_RUNTIME_TESTING_MENU is not set
-CONFIG_KSU=m
ERROR: savedefconfig does not match common/arch/arm64/configs/gki_defconfig
Target //common:kernel_aarch64_dist failed to build
Use --verbose_failures to see the command lines of failed build steps.
ERROR: /home/runner/work/KernelSU/KernelSU/android-kernel/common/BUILD.bazel:115:22 Middleman _middlemen/common_Skernel_Uaarch64_Udist-runfiles failed: (Exit 1): bash failed: error executing KernelConfig command (from target //common:kernel_aarch64_config) /bin/bash -c ... (remaining 1 argument skipped)
INFO: Elapsed time: 32.620s, Critical Path: 8.71s
INFO: 412 processes: 396 internal, 15 linux-sandbox, 1 local.
ERROR: Build did NOT complete successfully
ERROR: Build failed. Not running target
Error: Process completed with exit code 1.
```

File: kernel/Makefile (hunk 1 above)
Comment (dabao1955) [reply kernel/Makefile:19]: > Maybe you can use and then modify the build-lkm.yml?`obj-$(CONFIG_KSU) += kernelsu.o`

maybe fixed

**Review by tiann — APPROVED** [review]

**Pull request merged**